assets/dns-app/hooks/entrypoint.sh

```diff
@@ -6,6 +6,9 @@ if [ $1 = "update" ]; then
     echo "Updating resources"
     rig upsert -f /var/lib/gravity/resources/dns.yaml
 
+    echo "Deleting coredns daemonset that has been replaced by a deployment"
+    rig delete ds/coredns-worker --resource-namespace=kube-system --force
```

Comment:
When this executes during an upgrade, workers will be falling back to coredns pods running on masters before the new deployment comes up, correct?

Reply:
It might race a bit. The deployment is created first on line 6, but will only wait for the autoscaler to be scheduled, as we don't know here if the cluster has workers. Once the autosacler starts, it'll scale up the deployment appropriately. 

Whether the delete executes before the deployment pods get created I'm not sure, which would mean timeouts/errors to the worker service, and then yes fallback to the master service until the pods come up.